lib/src/geometry/collision_detection.dart

```diff
@@ -1,22 +1,24 @@
 import '../../extensions.dart';
 import '../components/mixins/collidable.dart';
 
+int _collidableTypeCompare(Collidable a, Collidable b) {
+  return a.collidableType.index - b.collidableType.index;
+}
+
 /// Check whether any [Collidable] in [collidables] collide with each other and
 /// call their onCollision methods accordingly.
 void collisionDetection(List<Collidable> collidables) {
+  collidables.sort(_collidableTypeCompare);
```

Comment:
Does sort cause unnecessary performance consumption?

Reply:
It should be very efficient since the list will almost always be sorted or close to sorted, and it makes the rest of the logic a lot more efficient.